src/core/Akka.Persistence/Journal/EventAdapters.cs

```diff
@@ -164,13 +164,22 @@ private static EventAdapters Create(ExtendedActorSystem system, IDictionary<stri
 
             // bindings is a enumerable of key-val representing the mapping from Type to handler.
             // It is primarily ordered by the most specific classes first, and secondly in the configured order.
+
             var bindings = Sort(adapterBindings.Select(kv =>
             {
-                var type = Type.GetType(kv.Key);
-                var adapter = kv.Value.Length == 1
-                    ? handlers[kv.Value[0]]
-                    : new CombinedReadEventAdapter(kv.Value.Select(h => handlers[h]));
-                return new KeyValuePair<Type, IEventAdapter>(type, adapter);
+                try
+                {
+                    var type = Type.GetType(kv.Key, true);
+                    var adapter = kv.Value.Length == 1
+                            ? handlers[kv.Value[0]]
+                            : new CombinedReadEventAdapter(kv.Value.Select(h => handlers[h]));
+                    return new KeyValuePair<Type, IEventAdapter>(type, adapter);
+                }
+                catch (Exception ex)
+                {
+                    var msgException = string.Format("Could not find type '{0}'", kv.Key);
+                    throw new TypeLoadException(msgException, ex);
```

Comment:
Something like "Could not find event adapter of type '{0}'". It's more informative, but this one is still good.